mitodl/bootcamp-ecommerce
# Add checkout data API

#### What are the relevant tickets?
Fixes #492 

#### What's this PR do?
Adds a new API `/api/checkout` to provide checkout data regarding applications, what they cost and what the user has already paid. There is an optional query parameter `application=<pk>` which will filter the list to just the application listed, if it exists and if it's awaiting payment.

#### How should this be manually tested?
Go to `/api/checkout` and verify that the data looks appropriate. If you add `?application=<pk>` to the url it should show only that application. Also, make sure nothing has changed in the payment page.

#### Context
This is intended to replace `/api/v0/bootcamps` in the near future, so there is a lot of overlapping functionality. In a future PR we will remove that API and migrate frontend logic to use this API instead.
